.github/workflows/run_periodic_tests.yml

```diff
@@ -114,42 +114,54 @@ jobs:
         if: matrix.os == 'ubuntu-latest'
         run: python -m nox -s scripts
 
-  #M-series Mac Mini
+  # M-series Mac Mini
   build-apple-mseries:
     if: github.repository_owner == 'pybamm-team'
     needs: style
     runs-on: [self-hosted, macOS, ARM64]
     env:
       GITHUB_PATH: ${PYENV_ROOT/bin:$PATH}
+      LD_LIBRARY_PATH: $HOME/.local/lib
     strategy:
       fail-fast: false
       matrix:
         python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
 
     steps:
       - uses: actions/checkout@v4
-      - name: Install python & create virtualenv
+      - name: Install Python & create virtualenv
         shell: bash
         run: |
           eval "$(pyenv init -)"
           pyenv install ${{ matrix.python-version }} -s
           pyenv virtualenv ${{ matrix.python-version }} pybamm-${{ matrix.python-version }}
 
-      - name: Install dependencies & run unit tests for Windows and MacOS
+      - name: Install build-time dependencies & run unit tests for M-series macOS runner
         shell: bash
+        env:
+          # Point scikits.odes to the correct SUNDIALS installation
+          SUNDIALS_INST: $HOME/.local/lib
+          # Homebrew environment variables
+          HOMEBREW_NO_INSTALL_CLEANUP: 1
+          NONINTERACTIVE: 1
         run: |
           eval "$(pyenv init -)"
           pyenv activate pybamm-${{ matrix.python-version }}
-          python -m pip install --upgrade pip wheel setuptools nox
+          python -m pip install --upgrade pip nox
+          # Don't use Homebrew to install SUNDIALS because scikits.odes looks for
+          # in Homebrew folders instead, which we don't want
```

Comment:
To clarify, using the Homebrew installation of SUNDIALS in this scenario **isn't a problem** (see #3511), but this workflow requires that we build the IDAKLU solver's build-time requirements from source, and therefore it is better to use just one SUNDIALS installation, i.e., the one present in `~/.local/`.

Even though we _can_ use `pybamm_install_odes` to set up `scikits.odes` on developer installations, we shouldn't do so because we limit its use case to user installations (and also because `noxfile.py` doesn't currently have it).